File: solutions/ruby/matrix/1/matrix.rb
class Matrix
  def initialize(matrix)
    @matrix = matrix
  end

  def rows
    @matrix.split(/\n/).map { |row| row.split.map(&:to_i) }
  end

  def columns
    columns = []
    
    rows.map(&:size).max.times do |i|
      columns << rows.map { |row| row[i] }
    end

    columns
  end
end

File: solutions/python/matrix/1/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.rows = list(map(lambda x: x.split(), matrix_string.split("\n")))
        self.columns = [int(row[i]) for i in range(len(self.rows[0])) for row in self.rows]

    def row(self, index):
        return list(map(int, self.rows[index-1]))

    def column(self, index):
        return [self.columns[i:i+len(self.rows)] for i in range(0, len(self.columns), len(self.rows))][index-1]
